src/main/java/org/prebid/server/bidder/ttx/TtxBidder.java

```diff
@@ -1,64 +1,216 @@
+    private final JacksonMapper mapper;
 
     public TtxBidder(String endpointUrl, JacksonMapper mapper) {
-        super(endpointUrl, RequestCreationStrategy.SINGLE_REQUEST, ExtImpTtx.class, mapper);
+        this.endpointUrl = HttpUtil.validateUrl(Objects.requireNonNull(endpointUrl));
+        this.mapper = Objects.requireNonNull(mapper);
     }
 
     @Override
-    protected void modifyRequest(BidRequest bidRequest, BidRequest.BidRequestBuilder requestBuilder,
-                                 List<ImpWithExt<ExtImpTtx>> impsWithExts) {
-        final List<Imp> modifiedImps = impsWithExts.stream()
-                .map(ImpWithExt::getImp)
-                .collect(Collectors.toList());
-        final Imp firstImp = modifiedImps.get(0);
-        final ExtImpTtx firstImpExt = impsWithExts.get(0).getImpExt();
-
-        final String zoneId = firstImpExt.getZoneId();
-        final TtxImpExt ttxImpExt = TtxImpExt.of(
-                TtxImpExtTtx.of(firstImpExt.getProductId(), StringUtils.isNotBlank(zoneId) ? zoneId : null));
-
-        final Imp modifiedFirstImp = firstImp.toBuilder().ext(mapper.mapper().valueToTree(ttxImpExt)).build();
-
-        if (modifiedImps.size() == 1) {
-            requestBuilder.imp(Collections.singletonList(modifiedFirstImp));
-        } else {
-            final List<Imp> subList = modifiedImps.subList(1, modifiedImps.size());
-            final List<Imp> finalizedImps = new ArrayList<>(subList.size() + 1);
-            finalizedImps.add(modifiedFirstImp);
-            finalizedImps.addAll(subList);
-            requestBuilder.imp(finalizedImps);
+    public Result<List<HttpRequest<BidRequest>>> makeHttpRequests(BidRequest request) {
+        final List<BidderError> errors = new ArrayList<>();
+        if (request.getImp().get(0).getBanner() == null
+                && request.getImp().get(0).getVideo() == null) {
+            errors.add(BidderError.badInput("At least one of [banner, video] "
+                    + "formats must be defined in Imp. None found"));
+            return Result.withErrors(errors);
         }
-        requestBuilder.site(modifySite(bidRequest.getSite(), firstImpExt.getSiteId()));
+        final HttpRequest<BidRequest> httpRequest = createRequest(request, errors);
+
+        return Result.of(Collections.singletonList(httpRequest), errors);
     }
 
-    private static Site modifySite(Site site, String siteId) {
-        final Site.SiteBuilder siteBuilder = site == null ? Site.builder() : site.toBuilder();
-        return siteBuilder
-                .id(siteId)
+    private HttpRequest<BidRequest> createRequest(BidRequest request, List<BidderError> errors) {
+        final BidRequest.BidRequestBuilder requestToUpdate = request.toBuilder();
+        final Imp firstImp = request.getImp().get(0);
+        try {
+            final ExtImpTtx extImpTtx = parseImpExt(firstImp);
+            requestToUpdate.site(updateSite(request.getSite(), extImpTtx.getSiteId()));
```

Comment:
have a couple ideas here:
1. populate `updatedSite` and `updatedFirstImp` variables in try section.
2. when creating `modifiedRequest` - set all determined variables.